### DISCUSSION TITLE
PrepSCTFindMarkers argument missing?

### DISCUSSION THREAD

**Discussion by LeoVelazco** [satijalab/seurat#6013] [Q&A]
Hi, I have ran SCTransform, vst.flavor = "v2". However, I am gaving trouble running PrepSCTFindMarkers(). It says the function is not found. Is there a way around running PrepSCTFindMarkers? 

 Error in PrepSCTFindMarkers() : 
  could not find function "PrepSCTFindMarkers"

**Comment by saketkc** [comment]
What version of Seurat are you using?

**Reply by Sayyam-Shah** [reply]
Hi, i'm getting the below error when running sctransform (vst.flavor='v2'). Could it be because of the version of Seurat I'm using? I converted an andata object from scanpy to seurat using SeuratDisk. 

Error in median(umi@x) : 
  trying to get slot "x" from an object of a basic class ("matrix") with no slots

**Reply by saketkc** [reply]
It would be helpful if you can paste the full code here.

**Reply by Sayyam-Shah** [reply]
Hello, thank you for replying!
hsc <- LoadH5Seurat("both.h5seurat")
hsc <- SCTransform(hsc , vst.flavor='v2')

I then get the below error.
![image](https://user-images.githubusercontent.com/102439837/173889920-a2efd882-1da3-4cf0-b6a1-7f7b38a9a8ac.png)

**Reply by saketkc** [reply]
thanks! What is the output of `class(GetAssayData(hsc, assay="RNA", slot="counts"))`

**Reply by Sayyam-Shah** [reply]
Hello, the output is: 
![image](https://user-images.githubusercontent.com/102439837/173892966-58b650aa-16e3-4726-94d7-d1969664dd71.png)

**Reply by saketkc** [reply]
That looks a bit weird since it needs to be a `dgCMatrix`, but if you run the following, the rest of the workflow should work:

```
hsc <- SetAssayData(object = hsc, assay = "RNA", slot = "counts",  new.data = as(GetAssayData(hsc, assay="RNA", slot="counts"), "dgCMatrix"))

```

**Reply by Sayyam-Shah** [reply]
Thank you so much! It worked.
